renderdoc/os/win32/win32_shellext.cpp

```diff
@@ -362,18 +399,219 @@ struct RDCThumbnailProvider : public IThumbnailProvider, IInitializeWithStream
+    {
+      const byte *jpgbuf = m_Thumb.pixels;
+      if(jpgbuf == NULL)
+        return E_NOTIMPL;
+
+      int w = thumbwidth;
+      int h = thumbheight;
+      int comp = 3;
+      thumbpixels = jpgd::decompress_jpeg_image_from_memory(jpgbuf, (int)thumblen, &w, &h, &comp, 3);
+    }
+    else
+    {
+      const ResourceFormatType compType = m_ddsData.format.type;
+      const uint32_t decompressedStride = thumbwidth * 4;
+      const uint32_t decompressedBlockWidth = 16;    // in bytes (4 byte/pixel)
+      const uint32_t decompressedBlockMaxSize = 64;
+
+      ResourceFormatType ddsFormatType = ResourceFormatType::BC1;
+      int compressedBlockSize = 8;
+      switch(compType)
+      {
+        case ResourceFormatType::BC1:
+          ddsFormatType = ResourceFormatType::BC1;
+          compressedBlockSize = 8;
+          break;
+        case ResourceFormatType::BC2:
+          ddsFormatType = ResourceFormatType::BC2;
+          compressedBlockSize = 16;
+          break;
+        case ResourceFormatType::BC3:
+          ddsFormatType = ResourceFormatType::BC3;
+          compressedBlockSize = 16;
+          break;
+        case ResourceFormatType::BC4:    // red color only, 1 bytes/pixel
+          ddsFormatType = ResourceFormatType::BC4;
+          compressedBlockSize = 8;
+          break;
+        case ResourceFormatType::BC5:    // red, green color only, 2 bytes/pixel
+          ddsFormatType = ResourceFormatType::BC5;
+          compressedBlockSize = 16;
+          break;
+        case ResourceFormatType::BC6:    // other formats
+        case ResourceFormatType::BC7:
+        case ResourceFormatType::A8:
+        case ResourceFormatType::ASTC:
+        case ResourceFormatType::D16S8:
+        case ResourceFormatType::D24S8:
+        case ResourceFormatType::D32S8:
+        case ResourceFormatType::EAC:
+        case ResourceFormatType::ETC2:
+        case ResourceFormatType::PVRTC:
+        case ResourceFormatType::R10G10B10A2:
+        case ResourceFormatType::R11G11B10:
+        case ResourceFormatType::R4G4:
+        case ResourceFormatType::R4G4B4A4:
+        case ResourceFormatType::R5G5B5A1:
+        case ResourceFormatType::R5G6B5:
+        case ResourceFormatType::R9G9B9E5:
+        case ResourceFormatType::Regular:
+        case ResourceFormatType::S8:
+        case ResourceFormatType::Undefined:
+        case ResourceFormatType::YUV10:
+        case ResourceFormatType::YUV12:
+        case ResourceFormatType::YUV16:
+        case ResourceFormatType::YUV8: return E_NOTIMPL;
+      }
+      thumbwidth = m_ddsData.width;
+      thumbheight = m_ddsData.height;
+      // Delete buffers
+      delete[] m_ddsData.subdata;
+      delete[] m_ddsData.subsizes;
+
+      byte *decompressed =
+          new byte[thumbheight * thumbwidth * 4];    // Decompressed DDS, 4 byte/pixel
+
+      unsigned char decompressedBlock[decompressedBlockMaxSize];
+      unsigned char greenBlock[16];
+      auto compBlockStart = m_Thumb.pixels;
```

Comment:
Please don't use `auto` unless the type is unknown or extremely long-winded - pretty much only iterators and lambdas.

Reply:
Replaced with `const byte *`.